src/github/client/github-user-client.ts

```diff
@@ -58,6 +58,12 @@ export class GitHubUserClient extends GitHubClient {
 		return await this.get<Octokit.AppsListInstallationsForAuthenticatedUserResponse>("/user/installations");
 	}
 
+	public async getNumberOfReposForInstallation(installationId: number): Promise<AxiosResponse<Octokit.AppsListInstallationsForAuthenticatedUserResponse>> {
+		return await this.get<Octokit.AppsListInstallationsForAuthenticatedUserResponse>(`/user/installations/{installationId}/repositories`, {
```

Comment:
Why the shift from graphql? If we're using graphql on the installation method to get the number of repos, shouldn't we do the same here?

Reply:
The previous graphql implementation used `GithubInstallationClient`, which requires a JWT token, which is created using the private key of the installation/organization/subscription.
But in our case, the installation/organization/subscription has not been connected and yet is not saved in our DB. So, they don't have a privateKey. So, we can't use grahpql request cause we have no privateKey to create the JWT.
As a result we are getting a 401 error, cause theres a mismatch of the privateKeys.
Thats why used a different endpoint in `GithubUserClient`, cause the token here is generated during the oauth authentication happening in middleware and gives us the same result.

Reply:
Yeah yeah I get that. Using the user client isn't what I was calling out. I was asking why we switched from making a request via graphql to REST. GHES has a graphql api and the gh client supports it `this.graphqlUrl = `${baseUrl}/api/graphql`

Reply:
I wasn't sure if the `graphql` works with the `githubToken` here in `GithubUserClient`. And as I was looking into the Github docs, found this endpoint:[ user/installations/{id}/repositories](https://docs.github.com/en/rest/apps/installations#list-repositories-accessible-to-the-user-access-token). So just used this.

Reply:
This has been reverted.